docs/develop/create-sso-office-add-ins-nodejs.md

```diff
@@ -124,453 +122,477 @@ This article walks you through the process of enabling single sign-on (SSO) in a
-    * If the POST request does not error, then the response from Azure AD is converted to JSON and sent to the client. This JSON object has an `access_token` property to which Azure AD has assigned the access token to Microsoft Graph.
+    * It only requests the minimum scopes it needs, such as `files.read.all`.
+    * It uses the MSAL authHelper to perform the OBO flow in the call to `acquireTokenOnBehalfOf`.
 
     ```javascript
-        const stsDomain = 'https://login.microsoftonline.com';
-        const tenant = 'common';
-        const tokenURLSegment = 'oauth2/v2.0/token';
-
-        try {
-            const tokenResponse = await fetch(`${stsDomain}/${tenant}/${tokenURLSegment}`, {
-                method: 'POST',
-                body: formurlencoded(formParams),
-                headers: {
-                    'Accept': 'application/json',
-                    'Content-Type': 'application/x-www-form-urlencoded'
-                }
-            });
-            const json = await tokenResponse.json();
-
-            res.send(json);
-        }
-        catch(error) {
-            res.status(500).send(error);
-        }
-    }
+    const authHeader = req.headers.authorization;
+    let oboRequest = {
+      oboAssertion: authHeader.split(" ")[1],
+      scopes: ["files.read.all"],
+    };
+  
+    const cca = authHelper.getConfidentialClientApplication();
+    cca
+      .acquireTokenOnBehalfOf(oboRequest)
+      .then(async (response) => {
+        // TODO 14: Call Microsoft Graph to get list of filenames.
+      })
+      .catch((err) => {
+        // TODO 15: Handle any errors.
+      });
     ```
 
-1. Save and close the file.
+1. Replace `TODO 14` with the following code. About this code, note:
 
-### Create the route that will fetch the data from Microsoft Graph
-
-1. Open the file `app.js` in the root of the project. Just below the route for '/dialog.html', add the following route. This route is called by the `makeGraphApiCall` function that you created in an earlier step.
+    * It constructs the URL for the Microsoft Graph API call and the makes the call via the `getGraphData` function.
+    * It returns errors by sending an HTTP 500 response along with details.
+    * On success it returns the JSON with the filename list to the client.
 
     ```javascript
-    app.get('/getuserdata', async function(req, res, next) {
-        
-        // TODO 15: Send a request to the Microsoft Graph REST endpoint.
+    console.log(response);
 
-        // TODO 16: Trim excess information from the returned data and relay it
-        //          to the client.
-        
-    });
-    ```
+    // Minimize the data that must come from MS Graph by specifying only the property we need ("name")
+    // and only the top 10 folder or file names.
+    const rootUrl = "/me/drive/root/children";
 
-1. Replace `TODO 15` with the following. About this code, note:
+    // Note that the last parameter, params, is hardcoded. If you reuse this code in
```

Comment:
Should code comment this be pulled out and inserted as an Important or Caution alert? People tend to skip code comments when reading.
> [!CAUTION]
> Negative potential consequences of an action.

Reply:
I added a security note section at bottom of the document (there are several things to mention.)